File: php/ATM.php
<?php

do {
    $nama = readline("Masukkan nama Anda: ");

    if (!empty($nama)) {
        $valid = true;
    } else {
        echo "Nama tidak boleh kosong, coba lagi!\n\n";
        $valid = false;
    }
} while (!$valid);

$nim = readline("Masukkan NIM (angka saja): ");
$saldo = intval($nim);

do {
    echo "\n===== MENU ATM =====\n";
    echo "1. Cek Saldo\n";
    echo "2. Tarik Tunai\n";
    echo "3. Setor Tunai\n";
    echo "4. Transfer\n";
    echo "5. Keluar\n";
    echo "Pilih: ";
    $pilihan = intval(readline());

    switch ($pilihan) {
        case 1:
            echo "Saldo Anda : Rp $saldo\n";
            break;

        case 2:
            $jumlah = intval(readline("Jumlah tarik: "));
            if ($jumlah <= $saldo) {
                $saldo -= $jumlah;
                echo "Berhasil menarik uang\n";
            } else {
                echo "Saldo tidak cukup\n";
            }
            break;

        case 3:
            $jumlah = intval(readline("Jumlah setor: "));
            $saldo += $jumlah;
            echo "Berhasil setor tunai\n";
            break;

        case 4:
            $transfer = intval(readline("Jumlah transfer: "));
            if ($transfer <= $saldo) {
                $saldo -= $transfer;
                echo "Transfer berhasil\n";
            } else {
                echo "Saldo tidak cukup\n";
            }
            break;

        case 5:
            echo "Terima kasih...\n";
            break;

        default:
            echo "Pilihan tidak valid\n";
    }

} while ($pilihan != 5);

?>

File: C/CPP/ATM.cpp
#include <iostream>
using namespace std;

int main() {
    string nama;
    bool valid = false;

    while (!valid) {
        cout << "Masukkan nama Anda: ";
        getline(cin, nama);

        if (!nama.empty()) {
            valid = true;
        } else {
            cout << "Nama tidak boleh kosong, coba lagi!\n\n";
        }
    }

    string nim;
    cout << "Masukkan NIM (angka saja): ";
    cin >> nim;

    long saldo = stol(nim);

    int pilihan;
    long jumlah, transfer;

    do {
        cout << "\n===== MENU ATM =====\n";
        cout << "1. Cek Saldo\n";
        cout << "2. Tarik Tunai\n";
        cout << "3. Setor Tunai\n";
        cout << "4. Transfer\n";
        cout << "5. Keluar\n";
        cout << "Pilih: ";
        cin >> pilihan;

        switch (pilihan) {
            case 1:
                cout << "Saldo Anda : Rp " << saldo << endl;
                break;

            case 2:
                cout << "Jumlah tarik: ";
                cin >> jumlah;
                if (jumlah <= saldo) {
                    saldo -= jumlah;
                    cout << "Berhasil menarik uang\n";
                } else {
                    cout << "Saldo tidak cukup\n";
                }
                break;

            case 3:
                cout << "Jumlah setor: ";
                cin >> jumlah;
                saldo += jumlah;
                cout << "Berhasil setor tunai\n";
                break;

            case 4:
                cout << "Jumlah transfer: ";
                cin >> transfer;
                if (transfer <= saldo) {
                    saldo -= transfer;
                    cout << "Transfer berhasil\n";
                } else {
                    cout << "Saldo tidak cukup\n";
                }
                break;

            case 5:
                cout << "Terima kasih...\n";
                break;

            default:
                cout << "Pilihan tidak valid\n";
        }

    } while (pilihan != 5);

    return 0;
}